### PR TITLE
Adding logical And , Or ,  Xor operators , [Issue #19]

### PR DESCRIPTION
**Pull request by Bharath-KumarReddy**
In the Binary_Calculator I had added the proper functionality of Logical  And , Logical Or , Xor operators respectively.

Please consider this and merge it as I am very interested to contribute in hacktoberfest_23 : )

### CODE REVIEW

File: js/binaryCalculator.js [hunk 1]
```diff
@@ -8,36 +8,62 @@ function insert(num) {
...
             break;
-        case '-':
-            value = num1 - num2;
+        case '^':
+            resultValue = logicalXor(num1, num2); 
             break;
         default:
-            value = ''
+            resultValue = "Error please Enter correctly"; 
             break;
```
Comment (Alitindrawan24) [inline js/binaryCalculator.js:58]: This changes break some funcionality on binary calculator, can you fix it ?

**Comment by Bharath-KumarReddy** [comment]
yes sure I will look into this :)

**Comment by Bharath-KumarReddy** [comment]
switch (operator) {
        case '+':
            resultValue = num1 + num2;
            break;
        case '-':
            resultValue = num1 - num2;
            break;
        case '*':
            resultValue = num1 * num2;
            break;
        case '/':
            if (num2 === 0) {
                res.value = "Error"; 
                return;
            }
            resultValue = num1 / num2;
            break;
        case '&':
            resultValue = num1 & num2; 
            break;
        case '|':
            resultValue = num1 | num2; 
            break;
        case '^':
            resultValue = num1 ^ num2; 
            break;
        default:
            res.value = "Error"; 
            return;
    }

    res.value = decimalToBinary(resultValue);
}


I just now updated the code , It will work Properly ,Once please check : )

**Comment by Clou-D-Walker** [comment]
Mention your/the issue number in PR bro, it would be good.

**Comment by Bharath-KumarReddy** [comment]
yes sure : )

**Comment by Alitindrawan24** [comment]
After merging a PR, this PR contains a conflict. Can you fix it? @Bharath-KumarReddy

**Comment by Bharath-KumarReddy** [comment]
Yes sure : )
I had fixed the conflicts once please check : )

File: index.html [hunk 2]
```diff
@@ -28,8 +28,9 @@ <h1 class="title">Binary Calculator</h1>
         <td><input class="button" type="button" value="/" onclick="insert('/')"></td>
       </tr>
       <tr>
-        <td><input class="button" type="button" value=">>" onclick="insert('>')"></td>
-        <td><input class="button" type="button" value="<<" onclick="insert('<')"></td>
```
Comment (Alitindrawan24) [inline index.html:32]: Your update deleted this button, this update is come from another PR

File: js/binaryCalculator.js [hunk 3]
```diff
@@ -1,76 +1,74 @@
...
-  const parts = result
-    .split(/([\+\-\*\/&|^><])/)
-    .filter((part) => part.trim() !== "");
+    
+    const parts = result.split(/([\+\-\*\/&|^])/).filter(part => part.trim() !== '');
 
-  if (parts.length !== 3) {
-    res.value = "Invalid Input";
-    return;
-  }
```
Comment (Alitindrawan24) [inline js/binaryCalculator.js:21]: This also came from another PR

**Review by Alitindrawan24 — COMMENTED** [review]
I think you need rebase/pull from main branch to get the latest update and apply your changes

**Comment by Bharath-KumarReddy** [comment]
Yes , I had added all the codes and  I a made a new PR with commit message "Complete_Binary_Calculator".

It will perform all logical and , or , xor , left shift , right shift and Airthematic operations;

Once please check this and merge it , Kindly inform incase of further modifications : )

File: index.html [hunk 4]
```diff
@@ -34,9 +34,18 @@ <h1 class="title">Binary Calculator</h1>
...
+        <td><input class="button" type="button" value="^" onclick="insert('^')"></td>
         <td><input class="button" type="button" value=">>" onclick="insert('>')"></td>
+      </tr>
+      <tr>
         <td><input class="button" type="button" value="<<" onclick="insert('<')"></td>
+<<<<<<< HEAD
+
+=======
         <td><input class="button" type="button" value="C" onclick="clc()"></td>
+>>>>>>> upstream/master
```
Comment (Alitindrawan24) [inline index.html:48]: This line contain unnecessary text, can you fix it ?

File: js/binaryCalculator.js [hunk 5]
```diff
@@ -67,6 +76,18 @@ function eql() {
...
+    return "0";
+  }
+  let binary = [];
+  while (num > 0) {
+    binary.push(num % 2);
+    num = Math.floor(num / 2);
+  }
+  return binary.reverse().join("");
+}
+=======
```
Comment (Alitindrawan24) [inline js/binaryCalculator.js:90]: This line too

**Pull request closed without merging**